python/cugraph-pyg/cugraph_pyg/sampler/cugraph_sampler.py

```diff
@@ -134,112 +231,87 @@ def _sampler_output_from_sampling_results(
-            raise ValueError("Edge src/dst type must match for homogeneous graphs")
-        row_dict = {
-            edge_type: torch.as_tensor(sampling_results.sources.values, device="cuda"),
-        }
-        col_dict = {
-            edge_type: torch.as_tensor(
-                sampling_results.destinations.values, device="cuda"
+    # Calculate nodes of interest based on unique nodes in order of appearance
+    # Use hop 0 sources since those are the only ones not included in destinations
+    # Use torch.concat based on benchmark performance (vs. cudf.concat)
+
+    if sampling_results_hop_0 is None:
+        sampling_results_hop_0 = sampling_results.iloc[
+            0 : (hops[1] if len(hops) > 1 else len(sampling_results))
+        ]
+
+    nodes_of_interest = (
+        cudf.Series(
+            torch.concat(
+                [
+                    torch.as_tensor(sampling_results_hop_0.sources, device="cuda"),
+                    torch.as_tensor(sampling_results.destinations, device="cuda"),
+                ]
             ),
-        }
-    else:
-        # Calculate nodes of interest based on unique nodes in order of appearance
-        # Use hop 0 sources since those are the only ones not included in destinations
-        # Use torch.concat based on benchmark performance (vs. cudf.concat)
-        nodes_of_interest = (
-            cudf.Series(
-                torch.concat(
-                    [
-                        torch.as_tensor(
-                            sampling_results_hop_0.sources.values, device="cuda"
-                        ),
-                        torch.as_tensor(
-                            sampling_results.destinations.values, device="cuda"
-                        ),
-                    ]
-                ),
-                name="nodes_of_interest",
-            )
-            .drop_duplicates()
-            .sort_index()
+            name="nodes_of_interest",
         )
-        del sampling_results_hop_0
+        .drop_duplicates()
+        .sort_index()
+    )
 
-        # Get the grouped node index (for creating the renumbered grouped edge index)
-        noi_index = graph_store._get_vertex_groups_from_sample(
-            torch.as_tensor(nodes_of_interest.values, device="cuda")
-        )
-        del nodes_of_interest
+    # Get the grouped node index (for creating the renumbered grouped edge index)
+    noi_index = graph_store._get_vertex_groups_from_sample(
+        torch.as_tensor(nodes_of_interest, device="cuda")
+    )
+    del nodes_of_interest
 
-        # Get the new edge index (by type as expected for HeteroData)
-        # FIXME handle edge ids/types after the C++ updates
-        row_dict, col_dict = graph_store._get_renumbered_edge_groups_from_sample(
-            sampling_results, noi_index
-        )
+    # Get the new edge index (by type as expected for HeteroData)
+    # FIXME handle edge ids/types after the C++ updates
```

Comment:
Are c++ updates are in another PR? or scheduled? What does notwork in the meantime?

Reply:
Future PR that I am currently working on.  Right now everything works, but we don't have all the promised features for version 23.10 that take advantage of the C++ MFG creation and format conversion.